### ISSUE TITLE
Navigation Rules to Enable Steps and Combine Multiple Navigation Rules

### ISSUE THREAD

**Issue by PSchmiedmayer** [ResearchKit/ResearchKit#1523]
Dear ResearchKit team,

We are using ResearchKit in the [ResearchKitOnFHIR](https://github.com/CardinalKit/ResearchKitOnFHIR) project to translate FHIR questionnaires into ReserachKit surveys.
In this process, we are required to translate FHIR conditionals to ReserachKit navigation rules. An easier translation of FHIR to ResearchKit would be enabled by the possibility of explicitly displaying steps on a condition rather than skipping steps based on specific conditions. We currently manually negate the statements in [`NavigationRules.swift`](https://github.com/CardinalKit/ResearchKitOnFHIR/blob/main/Sources/ResearchKitOnFHIR/FHIRToResearchKit/NavigationRules.swift).

We think this feature would also be helpful beyond the ReserachKitOnFHIR application to make it easier for developers to reason about their surveys using both ways to hide and display survey steps explicitly.

In addition, we have been trying to translate concatenated AND and OR operations to enable ReserachKit to evaluate multiple predicates when displaying a step. However, we have been unable to combine the predicates and let them be evaluated by ReserachKit. It would be great if you can help us by pointing us in the right direction on this feature and how concatenated predicates are supported by ResearchKit!

Thank you for your help and support in maintaining the project!

CC @vishnuravi

**Comment by louie-apple** [comment]
Hi @PSchmiedmayer and @vishnuravi: 

Thank you for mentioning this, we appreciate the feedback. Your suggestion of displaying steps based on conditions is interesting, and we are considering taking a look at this.
However we want to unblock you and tackle your main issue first, but we were unable to reproduce the concatenated AND and OR operations  issue.
Do you mind sharing with us a specific Predicate that is failing? That will help us understand your issue better.
Thank you!

**Comment by vishnuravi** [comment]
Hi @louie-apple,

Thank you for taking the time to look into the issue and confirming that it isn't a ResearchKit error. Upon further debugging, we found an issue in the way we were constructing predicates from FHIR enableWhen conditions. We were able to resolve this issue (https://github.com/StanfordBDHG/ResearchKitOnFHIR/pull/52). 

It would be great if, in the future, there were a simpler way to apply multiple conditions that enable steps, but this is working for now!

Thanks,
Vishnu and Paul
@StanfordBDHG
